app/parse.py

```diff
@@ -8,5 +15,40 @@
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+COURSE_FIELDS = [field.name for field in fields(Course)]
+
+
+def parse_single_course(page_soup: BeautifulSoup) -> Course:
+    return Course(name=page_soup.select_one("h3").text,
+                  short_description=page_soup.select_one(
+                      ".typography_landingTextMain__Rc8BD.mb-32"
+                  ).text,
+                  duration=page_soup.select_one(
+                      "p.typography_landingTextMain__Rc8BD.ProfessionCardTags_regularTag__yTc6K:last-of-type > span:last-of-type"
+                  ).text
+                  )
+
+
+def get_all_courses() -> [Course]:
+    page = requests.get(BASE_URL).content
+    page_soup = BeautifulSoup(page, "html.parser")
+    courses = page_soup.select(".ProfessionCard_cardWrapper__JQBNJ")
+
+    return [parse_single_course(course_soup) for course_soup in courses]
+
+
+def write_products_to_csv(courses: [Course]) -> None:
+    with open(OUTPUT_CSV_PATH, "w") as file:
```

Comment:
It's a good practice to specify the encoding when opening a file for writing, especially when dealing with web data that may contain special characters. Consider using `open(OUTPUT_CSV_PATH, 'w', encoding='utf-8')`.